### ISSUE TITLE
Constrained linear regression using Tensorflow

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#4068]
Constrained linear regression using Tensorflow<br>
<br>
http://bit.ly/2wfj6ZK<br>
<br>
About 0 Minutes<br>
<br>
<div>
<div>
<p>In this post, we will perform a constrained linear regression using Tensorflow. In linear regression, we create a model of the form $Y = W.X + b$. In this post, we will constrain $W$ and $b$ to be non-negative quantities. Such constraints are often useful in different applications. Say, if we want to model household total energy as the air conditioner energy plus some additive noise, then the contribution of both these quantities should be non-negative in nature.</p>
<p>We would be using gradient descent for our regression problem. We calculate the gradients wrt W and b, as usual: $$ \begin{eqnarray} \delta W = \frac{\delta cost}{\delta W} \\ \delta b = \frac{\delta cost}{\delta b} \\ \end{eqnarray} $$</p>
<p>The gist of our approach is that instead of updating as: $$ \begin{eqnarray} W = W - \delta W \times learning rate\\ b = b - \delta b \times learning rate\\ \end{eqnarray} $$</p>
<p>we update as:</p>
<p>$$ \begin{eqnarray} W = max(0, W - \delta W \times learning rate)\\ b = max(0, b - \delta b \times learning rate)\\ \end{eqnarray} $$</p>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via nipunbatra.github.io http://bit.ly/2Eq3lnf<br>
<br>
May 23, 2019 at 02:34PM
